src/schema_generators/generators.cljc

```diff
@@ -96,6 +98,10 @@
   "A mapping from schemas to generating functions that should be used."
   (s/=> (s/maybe Generator) Schema))
 
+(def Options
+  "A map of options to interact with core generator behaviors"
+  {:max-retries s/Int})
```

Comment:
Might it make sense to make this a function from Schema => max-retries, similar to how the other options are provided to `generator`?

Reply:
Do you think that instead of providing a `fn` to perform `Schema => max-retries`, the max-retries parameter could be part of the schema metadata and the code could lookup for it and use if found?

Reply:
Something like:

```clj
(let [leaf-generators (default-leaf-generators leaf-generators)
         gen (fn [s params]
               ((or (wrappers s) identity)
                (or (leaf-generators s)
                    (composite-generator (s/spec s) params))))]
     (generators/fmap
       (s/validator schema)
       (gen schema {:subschema-generator gen
                    :cache #?(:clj (java.util.IdentityHashMap.)
                              :cljs (atom {}))
                    :max-retries (retries s)})))

(s/defn retries :- s/Int
  [s  :- Schema]
  (or (-> s meta :max-retries)
      10))
```

Reply:
We could do that, but: 
 - You might not control the schema but still want to adjust the generator properties for something inside. 
 - All of the existing parameters to `generator` are functions from Schema to something, so I have a pretty strong bias to keeping this consistent with that unless there's a strong reason to deviate.

Reply:
Changes on https://github.com/plumatic/schema-generators/pull/29/commits/7043572018c08c519a1fa712d9f3ef7834acb3b6 and https://github.com/plumatic/schema-generators/pull/29/commits/3bc12dcfb35437788737f4b5dbf311d805fd4e2f